packages/calypso-products/src/is-titan-mail.ts

```diff
@@ -1,7 +1,10 @@
 import { camelOrSnakeSlug } from './camel-or-snake-slug';
-import { TITAN_MAIL_MONTHLY_SLUG } from './constants';
+import { TITAN_MAIL_MONTHLY_SLUG, TITAN_MAIL_YEARLY_SLUG } from './constants';
 import type { WithCamelCaseSlug, WithSnakeCaseSlug } from './types';
 
 export function isTitanMail( product: WithCamelCaseSlug | WithSnakeCaseSlug ): boolean {
-	return camelOrSnakeSlug( product ) === TITAN_MAIL_MONTHLY_SLUG;
+	return (
+		camelOrSnakeSlug( product ) === TITAN_MAIL_MONTHLY_SLUG ||
+		camelOrSnakeSlug( product ) === TITAN_MAIL_YEARLY_SLUG
+	);
 }
```

Comment:
This function isn't used in the rest of this PR - it may be simpler to push this out into its own PR so we can get it shipped.